substrate/frame/support/test/tests/enum_deprecation.rs

```diff
@@ -67,7 +67,11 @@ pub mod pallet {
 	#[pallet::storage_version(STORAGE_VERSION)]
 	pub struct Pallet<T>(_);
 
+	// Note that we use allow(deprecated) at the top of the file
+	// as deprecating the whole error is
+	// nonsensical when used in defining the runtime.
 	#[pallet::error]
+	#[deprecated = "example error"]
```

Comment:
maybe better to refactor this test without this deprecation and without the top level allow(deprecated)

Reply:
done